machines-coalgebras/src/Data/Machine/Mealy/Coalgebra/Monoidal.hs

```diff
@@ -0,0 +1,62 @@
+module Data.Machine.Mealy.Coalgebra.Monoidal
+  ( nudge,
+    nudgeLeft,
+    nudgeRight,
+    (/\),
+    (/+\),
+    (\/),
+  )
+where
+
+--------------------------------------------------------------------------------
+
+import Control.Category.Cartesian (split)
+import Data.Machine.Mealy.Coalgebra (MealyM (..))
+import Data.Profunctor (Profunctor (..), Strong (..))
+import Data.These (These)
+import Data.Trifunctor.Monoidal ((|*&&|), (|***|), (|*++|))
+
+--------------------------------------------------------------------------------
+
+-- | Given the sum of two bots, produce a bot who receives the sum of
+-- the inputs to the input bots and produces a wedge product of their
+-- outputs.
+nudge ::
+  (Monad m) => Either (MealyM m s i o) (MealyM m s i' o') -> MealyM m s (Either i i') (Maybe (Either o o'))
```

Comment:
@masaeedu Can I move this instance into `monoidal-functors` and define it generically?